implement check_offset for decorations (#13152)
minetest/minetest: doc/lua_api.txt

        spawn_by = "default:water",
        -- Node (or list of nodes) that the decoration only spawns next to.
        -- Checks the 8 neighboring nodes on the same height,
        -- and also the ones at the height plus the check_offset, excluding both center nodes.

>         check_offset = 0,
>         -- Specifies the offset that spawn_by should also check
>         -- The default value of -1 is usefull to e.g check for water next to the base node.
>         -- disables additional checking if 0, may have 1, 0, -1 as values.

Reviewer: ```suggestion
        check_offset = -,
        -- Specifies the offset that spawn_by should also check
        -- 0 disables additional checks, valid range: [-1, 1]
```

Author: A range to me implies a float value, hence my wording

Reviewer: well node positions are never floating point but I guess `valid values: {-1, 0, 1}` is clearer then

Author: I think the line mentioning -1 is the default value, and what it's use is should stay in any case, the use is not very clear and many apis /don't/ have their default values as the value set in the example of the documentation, so this would be confusing to me when not explicitly mentioned. I can reword it though if you'd like
